public/app/plugins/datasource/elasticsearch/query_def.ts

```diff
@@ -175,8 +187,10 @@ export const movingAvgModelSettings: any = {
 
 export function getMetricAggTypes(esVersion: any) {
   return _.filter(metricAggTypes, f => {
-    if (f.minVersion) {
-      return f.minVersion <= esVersion;
+    if (f.minVersion || f.maxVersion) {
+      const minVersion = f.minVersion || 0;
+      const maxVersion = f.maxVersion || esVersion;
+      return esVersion >= minVersion && esVersion <= maxVersion;
```

Comment:
> won't this return `true` if `esVersion` is configured as `70` ?

It depends... let's say we are comparing this metric agg type:

```
  {
    text: 'Moving Function',
    value: 'moving_fn',
    requiresField: false,
    isPipelineAgg: true,
    minVersion: 70,
  }
```
Then it will return `true` if `esVersion` is set to `70` but `false` if it's set to `60`. But if we have this meric agg type:
```
  {
    text: 'Moving Average',
    value: 'moving_avg',
    requiresField: false,
    isPipelineAgg: true,
    minVersion: 2,
    maxVersion: 60,
  }
```
Then it will return `false` if `esVersion` is set to `70` but `true` if it's set to `60`. I updated the tests to include this example to ensure it works as expected.

> also, maybe the upper bound should be exclusive as - assuming no breaking changes are introduced in a minor release - setting this to 70 would allow us to not care about any future 6.x release

I believe this works as described, do you have an example where this doesn't work?

Reply:
I wonder if we shouldn't move the function in the `_.filter` statement out of `getMetricAggTypes` so we can put together a bunch of different test scenarios to ensure it works as expected.

Reply:
>I believe this works as described, do you have an example where this doesn't work?

My bad, I misread the code.

> I wonder if we shouldn't move the function in the _.filter statement out of getMetricAggTypes so we can put together a bunch of different test scenarios to ensure it works as expected.

This is actually already being refactored (i'm personally already working on that as part of the React migration) so i wouldn't lose time on that.